### PR TITLE
Adds Click Listener to the pie chart slices

### PR DESCRIPTION
**Pull request by paulinembabu**
See Issue https://github.com/OpenSRP/opensrp-client-chw/issues/1240

### CODE REVIEW

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/listener/DefaultChwChartListenerFlv.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/viewholder/MyCommunityActivityDetailsViewHolder.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FragmentBaseActivity.java  2
- opensrp-chw/src/lmh/java/org/smartregister/chw/listener/ChwChartListenerFlv.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  6
- opensrp-chw/src/main/java/org/smartregister/chw/adapter/MyCommunityActivityDetailsAdapter.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  4
- opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java  2
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871&bid=18717304)

File: opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java [hunk 1]
```diff
@@ -171,4 +173,185 @@
 
         return result;
     }
+
+    @Contract(pure = true)
+    private static String getSql(String indicatorCode) {
```
Comment (keymane) [inline opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java:178]: Codacy found an issue: [Avoid really long methods.](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 11
           

Complexity increasing per file
==============================
- opensrp-chw/src/lmh/java/org/smartregister/chw/listener/ChwChartListenerFlv.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/listener/DefaultChwChartListenerFlv.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/viewholder/MyCommunityActivityDetailsViewHolder.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  6
- opensrp-chw/src/main/java/org/smartregister/chw/adapter/MyCommunityActivityDetailsAdapter.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FragmentBaseActivity.java  2
- opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  4
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871&bid=18717304)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+    @Test
+    public void testExtractRecordedLocationsReturnsEmptyArrayList() {
+        Mockito.doReturn(database).when(repository).getReadableDatabase();
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{
+                "location_id"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        List<String> locationList = ReportDao.extractRecordedLocations();
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(locationList.size(), 0);
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:61]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.*'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    public void testEligibleChildrenReportReturnsEmptyList() {
+        Mockito.doReturn(database).when(repository).getReadableDatabase();
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{
+                "base_entity_id"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        String communityId = "d5ff0ea1-bbc5-424d-84c2-5b084e10ef90";
+        Date dueDate = new DateTime().plusDays(7).toDate();
+        List<EligibleChild> eligibleChildren = ReportDao.eligibleChildrenReport(communityId, dueDate);
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(eligibleChildren.size(), 0);
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:88]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.*'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+import org.mockito.MockitoAnnotations;
+import org.smartregister.chw.core.domain.Person;
+import org.smartregister.chw.domain.EligibleChild;
+import org.smartregister.chw.domain.VillageDose;
+import org.smartregister.chw.model.FilterReportFragmentModel;
+import org.smartregister.chw.util.ReportingConstants;
+import org.smartregister.repository.Repository;
+
+import java.util.Date;
+import java.util.LinkedHashMap;
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:21]: Codacy found an issue: [Avoid unused imports such as 'java.util.LinkedHashMap'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+        Date dueDate = new DateTime().plusDays(7).toDate();
+        Map<String, Integer> map = new TreeMap<>();
+        map.put("ROTA", 0);
+        List<VillageDose> villageDoseList = ReportDao.villageDosesReport(villageName,communityId, dueDate);
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(villageDoseList.get(0).getRecurringServices(), map);
+    }
+
+    @Test
+    public void testMyCommunityActivityReportDetailsChildren12_59Dewormed() {
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:109]: Codacy found an issue: [The JUnit 4 test method name 'testMyCommunityActivityReportDetailsChildren12_59Dewormed' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+import org.smartregister.chw.util.ReportingConstants;
+import org.smartregister.repository.Repository;
+
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static org.junit.Assert.*;
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:26]: Codacy found an issue: [Avoid unused imports such as 'org.junit.Assert'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+        matrixCursor.addRow(new Object[]{"abdb6140-d54b-49d4-89e7-6f96c839c62b"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        String indicatorCode = ReportingConstants.ChildIndicatorKeys.COUNT_OF_CHILDREN_12_59_DEWORMED;
+        List<EligibleChild> eligibleChildren = ReportDao.myCommunityActivityReportDetails(indicatorCode);
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(eligibleChildren.get(0).getID(), "abdb6140-d54b-49d4-89e7-6f96c839c62b");
+    }
+
+    @Test
+    public void testMyCommunityActivityReportDetailsChildren12_59NotDewormed() {
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:122]: Codacy found an issue: [The JUnit 4 test method name 'testMyCommunityActivityReportDetailsChildren12_59NotDewormed' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+        Mockito.doReturn(database).when(repository).getReadableDatabase();
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{
+                "base_entity_id"});
+        matrixCursor.addRow(new Object[]{"abdb6140-d54b-49d4-89e7-6f96c839c62b"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        String communityId = "d5ff0ea1-bbc5-424d-84c2-5b084e10ef90";
+        Date dueDate = new DateTime().plusDays(7).toDate();
+        List<EligibleChild> eligibleChildren = ReportDao.eligibleChildrenReport(communityId, dueDate);
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(eligibleChildren.get(0).getID(), "abdb6140-d54b-49d4-89e7-6f96c839c62b");
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:75]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.*'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+import net.sqlcipher.database.SQLiteDatabase;
+
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.smartregister.chw.core.domain.Person;
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:13]: Codacy found an issue: [Avoid unused imports such as 'org.smartregister.chw.core.domain.Person'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+        matrixCursor.addRow(new Object[]{"ROTA 1"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        String villageName = "Grenaligne";
+        String communityId = "d5ff0ea1-bbc5-424d-84c2-5b084e10ef90";
+        Date dueDate = new DateTime().plusDays(7).toDate();
+        Map<String, Integer> map = new TreeMap<>();
+        map.put("ROTA", 0);
+        List<VillageDose> villageDoseList = ReportDao.villageDosesReport(villageName,communityId, dueDate);
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(villageDoseList.get(0).getRecurringServices(), map);
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:105]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.*'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.smartregister.chw.core.domain.Person;
+import org.smartregister.chw.domain.EligibleChild;
+import org.smartregister.chw.domain.VillageDose;
+import org.smartregister.chw.model.FilterReportFragmentModel;
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:16]: Codacy found an issue: [Avoid unused imports such as 'org.smartregister.chw.model.FilterReportFragmentModel'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 12]
```diff
@@ -0,0 +1,251 @@
...
+    @Test
+    public void testExtractRecordedLocations() {
+        Mockito.doReturn(database).when(repository).getReadableDatabase();
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{
+                "location_id"});
+        matrixCursor.addRow(new Object[]{"d5ff0ea1-bbc5-424d-84c2-5b084e10ef90"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        List<String> locationList = ReportDao.extractRecordedLocations();
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(locationList.get(0), "d5ff0ea1-bbc5-424d-84c2-5b084e10ef90");
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:50]: Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.*'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  4
- opensrp-chw/src/main/java/org/smartregister/chw/adapter/MyCommunityActivityDetailsAdapter.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FragmentBaseActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  6
- opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java  1
- opensrp-chw/src/lmh/java/org/smartregister/chw/listener/ChwChartListenerFlv.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/listener/DefaultChwChartListenerFlv.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/viewholder/MyCommunityActivityDetailsViewHolder.java  1
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871&bid=18717304)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+        Date dueDate = new DateTime().plusDays(7).toDate();
+        Map<String, Integer> map = new TreeMap<>();
+        map.put("ROTA", 0);
+        List<VillageDose> villageDoseList = ReportDao.villageDosesReport(villageName, communityId, dueDate);
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(villageDoseList.get(0).getRecurringServices(), map);
+    }
+
+    @Test
+    public void testMyCommunityActivityReportDetailsChildren12_59Dewormed() {
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:104]: Codacy found an issue: [The JUnit 4 test method name 'testMyCommunityActivityReportDetailsChildren12_59Dewormed' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

File: opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+        matrixCursor.addRow(new Object[]{"abdb6140-d54b-49d4-89e7-6f96c839c62b"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        String indicatorCode = ReportingConstants.ChildIndicatorKeys.COUNT_OF_CHILDREN_12_59_DEWORMED;
+        List<EligibleChild> eligibleChildren = ReportDao.myCommunityActivityReportDetails(indicatorCode);
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(eligibleChildren.get(0).getID(), "abdb6140-d54b-49d4-89e7-6f96c839c62b");
+    }
+
+    @Test
+    public void testMyCommunityActivityReportDetailsChildren12_59NotDewormed() {
```
Comment (keymane) [inline opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java:117]: Codacy found an issue: [The JUnit 4 test method name 'testMyCommunityActivityReportDetailsChildren12_59NotDewormed' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/viewholder/MyCommunityActivityDetailsViewHolder.java  1
- opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  6
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  4
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FragmentBaseActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/listener/DefaultChwChartListenerFlv.java  1
- opensrp-chw/src/lmh/java/org/smartregister/chw/listener/ChwChartListenerFlv.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/adapter/MyCommunityActivityDetailsAdapter.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java  2
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871&bid=18717304)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/adapter/MyCommunityActivityDetailsAdapter.java  1
- opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/viewholder/MyCommunityActivityDetailsViewHolder.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  4
- opensrp-chw/src/main/java/org/smartregister/chw/listener/DefaultChwChartListenerFlv.java  1
- opensrp-chw/src/lmh/java/org/smartregister/chw/listener/ChwChartListenerFlv.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FragmentBaseActivity.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  6
- opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java  2
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871&bid=18717304)

File: opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+            bundle.putString(Constants.ReportParameters.INDICATOR_CODE, pieChartSlice.getKey());
+            FragmentBaseActivity.startMe(context, MyCommunityActivityDetailsFragment.TAG, context.getString(R.string.children), bundle);
+
+        } else {
+            Toast.makeText(context, pieChartSlice.getLabel(), Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    public interface Flavor {
+        Boolean showMyCommunityActivityReport();
```
Comment (rkodev) [inline opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java:35]: Move this method to the interface `ChwApplication.Flavor` you can delete all other implementation of the interface

Comment (rkodev) [reply opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java:35]: Also change the return type from `Boolean` to `boolean`

File: opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java [hunk 16]
```diff
@@ -0,0 +1,34 @@
+package org.smartregister.chw.listener;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.widget.Toast;
+
+import org.smartregister.chw.R;
+import org.smartregister.chw.activity.FragmentBaseActivity;
+import org.smartregister.chw.application.ChwApplication;
+import org.smartregister.chw.application.ChwApplicationFlv;
```
Comment (keymane) [inline opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java:10]: Codacy found an issue: [Avoid unused imports such as 'org.smartregister.chw.application.ChwApplicationFlv'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/adapter/MyCommunityActivityDetailsAdapter.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  6
- opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/viewholder/MyCommunityActivityDetailsViewHolder.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  4
- opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FragmentBaseActivity.java  2
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871&bid=18717304)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  4
- opensrp-chw/src/main/java/org/smartregister/chw/activity/FragmentBaseActivity.java  2
- opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/listener/ChwChartListener.java  2
- opensrp-chw/src/main/java/org/smartregister/chw/viewholder/MyCommunityActivityDetailsViewHolder.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/adapter/MyCommunityActivityDetailsAdapter.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  6
         

Clones added
============
- opensrp-chw/src/main/java/org/smartregister/chw/fragment/MyCommunityActivityDetailsFragment.java  1
- opensrp-chw/src/main/java/org/smartregister/chw/dao/ReportDao.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871&bid=18717304)

**Pull request merged**